shared/data/src/main/java/com/ivy/data/remote/impl/RemoteExchangeRatesDataSourceImpl.kt

```diff
@@ -1,28 +1,41 @@
 package com.ivy.data.remote.impl
 
 import arrow.core.Either
+import com.ivy.data.model.primitive.AssetCode
 import com.ivy.data.remote.RemoteExchangeRatesDataSource
 import com.ivy.data.remote.responses.ExchangeRatesResponse
 import io.ktor.client.HttpClient
 import io.ktor.client.call.body
 import io.ktor.client.request.get
+import java.util.Locale
 import javax.inject.Inject
 
 class RemoteExchangeRatesDataSourceImpl @Inject constructor(
     private val ktorClient: dagger.Lazy<HttpClient>,
 ) : RemoteExchangeRatesDataSource {
+    private val currencyCodeLowerMask: String = "###"
```

Comment:
Why do we need this? Let's hardcode the URLs to eur because EUR will always be available as a base and the code is simpler

Reply:
We need it because the URL Changes based on the currency Code 
eg ->  This is for EUR
https://cdn.jsdelivr.net/npm/@fawazahmed0/currency-api@latest/v1/currencies/eur.min.json 

If I want USD then 
https://cdn.jsdelivr.net/npm/@fawazahmed0/currency-api@latest/v1/currencies/usd.min.json

the API path itself changes rather than providing a query parameter hence we need this

Reply:
No we don't - just fetch EUR which I know it works in our free limited open source API which is btw some json files in a GitHub repo. Always fetch EUR -> recompute to user's base

Reply:
Got it